### PR TITLE
Faris/cleanup optimizations

### PR DESCRIPTION
**Pull request by farisashai**
some cleanup and optimizations for repo

- serversideprops should return as little data as possible, moved event slicing to serverside since the slice range is already hardcoded in
- shallow navigate page routes with native nextjs router to avoid recalculating serversideprops
- default event page to all time so the page isn't empty on initial load ever
- format event links without https:// correctly instead of assuming relative links
- fix hydration error trying to run the browser blob api serverside pre-rendering
- load gifs correctly in top 3 user cards

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -69,15 +70,17 @@ const PortalHomePage = ({
...
+  const router = useRouter();
+
   useEffect(() => {
     if (checkInResponse) {
       // In dev mode, this runs twice because of reactStrictMode in nextConfig.
       // This will only be run once in prod or deployment.
       processCheckInResponse(checkInResponse);
       // Clear the query params without re-triggering getServerSideProps.
-      window.history.replaceState(null, '', '/');
+      router.push(`${config.homeRoute}`, undefined, { shallow: true });
```
Comment (alexzhang1618) [inline src/pages/index.tsx]: Does this work? in my testing, after you check-in with the query params and `{ shallow: true }` re-renders w/o reloading props so the toast gets triggered infinitely

Comment (farisashai) [reply src/pages/index.tsx]: idk i will remove

File: src/pages/leaderboard.tsx [hunk 2]
```diff
@@ -111,7 +111,7 @@ const LeaderboardPage = ({ sort, leaderboard, user: { uuid } }: LeaderboardProps
           ]}
           value={sort}
           onChange={sort => {
-            router.push(`${config.leaderboardRoute}?sort=${sort}`);
+            router.push(`${config.leaderboardRoute}?sort=${sort}`, undefined, { shallow: true });
```
Comment (alexzhang1618) [inline src/pages/leaderboard.tsx]: this bricks the leaderboard on the preview

**Review by alexzhang1618 — APPROVED** [review]
lgtm 😁

**Pull request merged**